scl/cisco/plugin.conf

```diff
@@ -37,26 +37,11 @@
 
 #
 # parses a cisco timestamp with explicit date-parser
-# It ignores msec and year information
+# It ignores timezone information
 #
 block parser cisco-timestamp-parser(template()) {
     channel {
-        if {
-            # timestamp from Cisco Unified Call Manager, example "Jun 14 11:57:27 PM.685 UTC"
-            # NOTE: drops msecs and timezone
-            filter { match('^[.*]?([A-Za-z]{3} [0-9 ]\d \d{2}:\d{2}:\d{2} ((AM)|(PM)))' template(`template`) flags(store-matches)); };
-            parser { date-parser(format('%b %d %H:%M:%S %p') template("$1")); };
-        } elif {
-            # timestamp without AM/PM mark, example: "Apr 29 13:58:40.411"
-            # NOTE: drops msecs and timezone
-
-            filter { match('^[.*]?([A-Za-z]{3} [0-9 ]\d \d{2}:\d{2}:\d{2})' template(`template`) flags(store-matches)); };
-            parser { date-parser(format('%b %d %H:%M:%S') template("$1")); };
-        } else {
-            # timestamp with year information, example: "Apr 29 2017 13:58:40.411"
-            filter { match('^[.*]?([A-Za-z]{3} [0-9 ]\d \d{4} \d{2}:\d{2}:\d{2})' template(`template`) flags(store-matches)); };
-            parser { date-parser(format('%b %d %Y %H:%M:%S') template("$1")); };
-        };
+        parser { date-parser(format(' *%b %d %H:%M:%S %p.%f', ' *%b %d %H:%M:%S.%f', ' *%b %d %Y %H:%M:%S.%f') template(`template`)); };
     };
```

Comment:
Much much better! thank you. btw, we could theoretically add all of these formats with a %Z that should parse the timezone as well. something like:

```
parser { date-parser(format(' *%b %d %H:%M:%S %p.%f %Z', ' *%b %d %H:%M:%S %p.%f', ' *%b %d %H:%M:%S.%f %Z', ' *%b %d %H:%M:%S.%f', ' *%b %d %Y %H:%M:%S.%f %Z', ' *%b %d %Y %H:%M:%S.%f') template(`template`)); };
```

Although %Z in our current strptime() implementation is pretty limited (it only recognizes UTC, GMT and the name of the local timezone), that can be extended later, still I think it is better to add it here, and potentially improve strptime() at a later point.

Consider this optional, I already think this is a huge improvement.

Reply:
btw , why do you have a space in front? we didn't have that in the previous implementation.

Reply:
For the extra space, this was probably an error on my side when juggling with this branch and the one where I was trying to fix the test suite on my test machine running FreeBSD.

Adding the time zone, I spot another issue related to 12-hour clock formats and hopefully fixed it (H -> %I).  BTW, do you have some cisco logs with time zone information to share to extend the test suite?